app-backend/tiler/src/main/scala/TilerRoutes.scala

```diff
@@ -0,0 +1,121 @@
+package com.azavea.rf.tiler
+
+import akka.http.scaladsl.model.{ContentType, HttpEntity, HttpResponse, MediaTypes}
+import com.azavea.rf.tiler.render._
+import com.typesafe.scalalogging.LazyLogging
+import geotrellis.raster._
+import geotrellis.raster.equalization.HistogramEqualization
+import geotrellis.raster.histogram.Histogram
+import geotrellis.raster.render.{ColorRamps, ColorRamp, ColorMap, Png}
+import geotrellis.raster.sigmoidal.SigmoidalContrast
+import geotrellis.spark._
+import akka.http.scaladsl.server._
+import akka.http.scaladsl.server.Directives._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.Try
+
+trait TilerRoutes extends LazyLogging {
+
+  def subsetBands: Directive1[Option[Array[Int]]] =
+    parameters('bands.?).as { bands: Option[String] =>
+      for { b <- bands }
+        yield b.split(",").map(_.toInt)
+    }
+
+  def sigmoidalContrast: Directive1[Option[MultibandTile => MultibandTile]] =
+    parameters('alpha.as[Double].?, 'beta.as[Double].?).as { (alpha: Option[Double], beta: Option[Double]) =>
+      for (a <- alpha; b <- beta) yield SigmoidalContrast(_: MultibandTile, a, b)
+    }
+
+  def histogramEqualization: Directive1[Option[(MultibandTile, Array[Histogram[Double]]) => MultibandTile]] =
+    parameters('equalize.as[Boolean].?(false)).as { equalize: Boolean =>
+      if (equalize) Some(HistogramEqualization(_: MultibandTile, _: Array[Histogram[Double]])) else None
+    }
+
+  def getColorRamp(str: String): Option[ColorRamp] = {
+    val chunks = str.split(",").map(_.trim)
+    if (chunks.length > 1)
+      Try(ColorRamp(chunks.map(_.toInt))).toOption
+    else
+      Option {
+        str match {
+          case "BlueToOrange" =>
+            ColorRamps.BlueToOrange
+          case "BlueToRed" =>
+            ColorRamps.BlueToRed
+          case "LightToDarkGreen" =>
+            ColorRamps.LightToDarkGreen
+          case _ => null
+        }
+      }
+  }
+
+  def render: Directive1[(MultibandTile, Array[Histogram[Double]]) => Png] =
+    parameters('render.as[String].?("grey"), 'breaks.as[String].?, 'ramp.as[String].?).as {
```

Comment:
I think it would be good to declare parameters somewhere common/consistent. It will make parsing routes easier I think.

One idea would be to parse parameters into a case class for each endpoint.